DEPENDENCIES.md

```diff
@@ -0,0 +1,71 @@
+# Upgrading Go Version and Dependencies in Kubeflow
+
+This guide outlines the steps to upgrade the Go version and dependencies in the Kubeflow project, including the necessary changes in the red-hat-data-services/kubeflow fork.
+
+## Upgrading Go Version
+
+Upgrading the Go version should be done in a separate PR to isolate the changes and make review easier.
+
+> [!IMPORTANT]  
+> Images are to be built in the [ubi8/go-toolset](https://catalog.redhat.com/software/containers/ubi8/go-toolset/5ce8713aac3db925c03774d1) container.
+> It contains a customized FIPS-compatible version of Go, that however lags behind the latest upstream Go version.
+> Always use a Go version that has a supporting go-toolset image available.
+
+1. Begin by reading [Go release notes](https://go.dev/doc/devel/release) to identify potential incompatibilities.
+
+2. Update the Go version in the following files:
+   - `components/**` `/` `go.mod`: Update the `go` directive at the top of the file.
+   - `components/**` `/` `Dockerfile`: Update the Go version in the base image.
+
+3. Run the following commands to update and verify the build:
+
+    ```shell
+    go mod tidy
+    make test
+    ```
+
+4. Commit these changes and create a pull request for the Go version upgrade.
+
+   > [!WARNING]  
+   > Use the `Manifest List Digest` and not the `Image Digest` when locating sha256 in the Red Hat Image Catalog entry.
+
+5. After merging the Go upgrade PR, update the fork at https://github.com/red-hat-data-services/kubeflow:
```

Comment:
Maybe we should say that this change should be done once the odh changes were synced with the rhds main branch too?

Reply:
Does not really matter, works fine independently, see CI on 

* https://github.com/red-hat-data-services/kubeflow/pull/447

Reply:
I understand, but do you want to do it in this order? :thinking:

Reply:
Any order that keeps CI green at all times is a good order imo

Reply:
btw, the change into the rhds will propagate only once the odh/main is synced with odh/v1.9-branch - if I'm not missing anything: https://github.com/red-hat-data-services/kubeflow/commit/36f270ee380f560a3838445961de510deb785196

But yeah... I don't insist... it's just that we may end up in a situation where in rhds we are already using newer version of golang for the dockerfiles but the rest of the code/dependencies are on the previous version - which may not be an issue most of the time, but sometimes it could, IDK.